src/ja/nicomanga/src/eu/kanade/tachiyomi/extension/ja/nicomanga/Nicomanga.kt

```diff
@@ -0,0 +1,167 @@
+
+    override fun latestUpdatesParse(response: Response): MangasPage = mangaListParse(response)
+
+    override fun popularMangaRequest(page: Int): Request {
+        val url = "$baseUrl/manga-list.html".toHttpUrl().newBuilder()
+            .addQueryParameter("page", page.toString())
+            .addQueryParameter("sort", "views")
+            .addQueryParameter("sort_type", "DESC")
+            .build()
+        return GET(url)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage = mangaListParse(response)
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/manga-list.html".toHttpUrl().newBuilder()
+            .addQueryParameter("page", page.toString())
+            .addQueryParameter("artist", "")
+            .addQueryParameter("author", "")
+            .addQueryParameter("group", "")
+            .addQueryParameter("m_status", "")
+            .addQueryParameter("name", query)
+            .addQueryParameter("genre", "")
+            .addQueryParameter("ungenre", "")
+            .addQueryParameter("magazine", "")
+            .addQueryParameter("sort", "last_update")
+            .addQueryParameter("sort_type", "DESC")
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage = mangaListParse(response)
+
+    override fun mangaDetailsParse(response: Response): SManga = SManga.create().apply {
+        val doc = Jsoup.parse(response.body.string())
+        author = doc.select("ul.manga-info a[href^=\"manga-author\"]").joinToString { it.text() }
+        genre = doc.select("ul.manga-info a[href^=\"manga-list-genre\"]").joinToString { it.text() }
+        val statusText = Nicomanga.statusRegex.find(doc.select(".manga-info li:has(i.fa-spinner) a").attr("href"))?.groupValues?.get(0) ?: ""
+        status = when (statusText) {
+            "on-going" -> {
+                SManga.ONGOING
+            }
+            "completed" -> {
+                SManga.COMPLETED
+            }
+            else -> {
+                SManga.UNKNOWN
+            }
+        }
+    }
+
+    override fun chapterListRequest(manga: SManga): Request {
+        val slug = Nicomanga.urlRegex.find(manga.url)?.groupValues?.get(0) ?: ""
+        return GET("$baseUrl/app/manga/controllers/cont.Listchapterapi.php?slug=$slug")
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val doc = Jsoup.parse(response.body.string())
+        val chapterList = doc.select("ul > a")
+        var lastNum = 0f
+        val chapters = chapterList.map { chapter ->
+            SChapter.create().apply {
+                name = chapter.attr("title").trim()
+                setUrlWithoutDomain(URL(URL(baseUrl), chapter.attr("href")).toString())
+                chapter_number = Nicomanga.floatRegex.find(chapter.attr("title").trim())?.groupValues?.get(0)?.toFloat() ?: (lastNum + 0.01f)
```

Comment:
Sometimes chapters are in the wrong order.

Reply:
Sometimes chapters are out of order

Reply:
Sadly not, since they correct chapters that are out of order after a few hours at most. So it doesn't concern the long term, but if you happen to refresh or load in that timeframe, you'd get wrong chapter numbers and when the order is rectified, it could lead to some issues.

Reply:
Please explain more clearly. I don't understand why out-of-order chapters justify your parsing chapter numbers manually.

Reply:
From the way other extensions work, I assume, that by default, chapters are numbered as they come from the source. If that isn't the case and Tachiyomi does parse the chapter number in the name by default aswell, it could indeed be removed here. Is that the case?